packages/fuel-gauge/src/transaction-summary.test.ts

```diff
@@ -621,5 +626,101 @@ describe('TransactionSummary', () => {
+    });
+
+    it('Should return contract created operations', async () => {
+      using launched = await launchTestNode();
+
+      const {
+        wallets: [wallet],
+      } = launched;
+
+      const tx = await MultiTokenContractFactory.deploy(wallet);
+      const result = await tx.waitForResult();
+
+      expect(result.transactionResult.operations).toHaveLength(1);
+      expect(result.transactionResult.operations[0].name).toEqual(OperationName.contractCreated);
+      expect(result.transactionResult.operations[0].from?.type).toEqual(1);
+      expect(result.transactionResult.operations[0].from?.address).toEqual(wallet.address.toB256());
+      expect(result.transactionResult.operations[0].to?.type).toEqual(0);
+      expect(result.transactionResult.isTypeCreate).toEqual(true);
+    });
+
+    it('should have no assets returned for contract call operations', async () => {
+      using launched = await launchTestNode({
+        contractsConfigs: [
+          {
+            factory: TokenContractFactory,
+          },
+        ],
+      });
+
+      const {
+        wallets: [wallet],
+        contracts: [contract],
+      } = launched;
+
+      const tx = await contract.functions.mint_coins(100).call();
+      const result = await tx.waitForResult();
+
+      expect(result.transactionResult.operations).toHaveLength(1);
+      expect(result.transactionResult.operations[0].name).toEqual(OperationName.contractCall);
+      expect(result.transactionResult.operations[0].from?.type).toEqual(1);
+      expect(result.transactionResult.operations[0].from?.address).toEqual(wallet.address.toB256());
+      expect(result.transactionResult.operations[0].to?.address).toEqual(contract.id.toB256());
+      expect(result.transactionResult.operations[0].to?.type).toEqual(0);
+      expect(result.transactionResult.operations[0].assetsSent).toBeUndefined();
+    });
+
+    it('should return the correct operations for a mint transaction', async () => {
+      using launched = await launchTestNode({
+        nodeOptions: {
+          args: ['--poa-instant', 'false', '--poa-interval-period', '1ms'],
+          loggingEnabled: false,
+        },
+      });
+      const { provider } = launched;
+
+      const txnInfo = await provider.getTransactions({ first: 1 });
+
+      const transactions = txnInfo.transactions[0];
+
+      expect(transactions.type).toEqual(TransactionType.Mint);
+      expect(transactions.txPointer?.blockHeight).toEqual(1);
+      expect(transactions.txPointer?.txIndex).toEqual(0);
+      expect(transactions.inputContract?.txID).toEqual(ZeroBytes32);
+      expect(transactions.inputContract?.balanceRoot).toEqual(ZeroBytes32);
+      expect(transactions.inputContract?.stateRoot).toEqual(ZeroBytes32);
+      expect(transactions.inputContract?.type).toEqual(TransactionType.Create);
+      expect(transactions.mintAmount?.eq(bn(0))).toBe(true);
+      expect(transactions.mintAssetId).toEqual(provider.getBaseAssetId());
+    });
```

Comment:
@maschad It seems that we are not validating the returned operations here

Reply:
You can use `getTransactionsSummaries` for this test case

Reply:
Because that's a `hidden` method, I didn't want to use it in the e2e test, but if you agree we should encourage consumers to use this method feel free to support that recommendation on [this discussion](https://github.com/FuelLabs/fuels-ts/discussions/2429) and I can update this test.

Reply:
@maschad You can use it on the test, this method should not be hidden

Reply:
I don't think it can be used in this test as a [mint transaction](https://docs.fuel.network/docs/specs/tx-format/transaction/#transactionmint) has no owner (since it's not signed) and `getTransactionSummaries` requires an `owner` [as a param in the filters](https://github.com/FuelLabs/fuels-ts/blob/mc/chore/add-integration-tests-operations/packages/account/src/providers/transaction-summary/get-transaction-summary.ts#L130).

Reply:
@maschad I think we can remove this test. The mint operation is intended to describe an asset mint operation. However, the SDK is not assembling this operation. I've created an [issue](https://github.com/FuelLabs/fuels-ts/issues/3008) to review all operations that are not being used.

Reply:
That's a good point, and upon closer review we already have decoding tests for it 

7bdc98787